src/react-native/metro.ts

```diff
@@ -20,33 +20,61 @@ import * as recast from 'recast';
 import x = recast.types;
 import t = x.namedTypes;
 import chalk from 'chalk';
+import { PackageDotJson, hasPackageInstalled } from '../utils/package-json';
+import { getExpoMinimalMetroConfigFileContent } from './expo.vendor';
 
 const b = recast.types.builders;
 
 const metroConfigPath = 'metro.config.js';
 
-export async function patchMetroConfig() {
-  const mod = await parseMetroConfig();
-
+export async function patchMetroConfig(
+  packageJson: PackageDotJson,
+  isExpoManagedProject: boolean,
+) {
   const showInstructions = () =>
     showCopyPasteInstructions(metroConfigPath, getMetroConfigSnippet(true));
 
+  const doesConfigExist = fs.existsSync(metroConfigPath);
+
+  if (!doesConfigExist && isExpoManagedProject) {
+    return await createExpoMinimalMetroConfigWithSentry(metroConfigPath);
+  }
+
+  if (!doesConfigExist) {
+    Sentry.setTag('metro-config-path', 'not-found');
+    return await showInstructions();
+  }
```

Comment:
what if we invert the if condition? (I can't make suggestions because i believe im not part of this project
```typescript
  const missingMetroConfig = !fs.existsSync(metroConfigPath);

  if (missingMetroConfig && isExpoManagedProject) {
    return await createExpoMinimalMetroConfigWithSentry(metroConfigPath);
  }

  if (missingMetroConfig ) {
    Sentry.setTag('metro-config-path', 'not-found');
    return await showInstructions();
  }
```